Primary analysis interface a11y fixes

dannon opened the pull request:
WIP, see individual commits for details.  Started with popping some fixes out of an out of date a11y testing configuration branch I've been working on (using playwright to run axe and spit reports for different areas in the app -- still forthcoming), and addressing some new stuff manually.  Targeting `WCAG 2.1 AAA`

Targeting primary analysis interface first in this pass, just the initial page load with a real(ish) history.

Before:
| Severity | Count|
| ------------- | ------------- |
| Critical | 42  |
| Serious  | 7  |
| Moderate | 61  |
| Minor | 0  |

After:
| Severity | Count|
| ------------- | ------------- |
| Critical | 0  |
| Serious  | 0  |
| Moderate | 1  |
| Minor | 0  |



The only issue left for full aaa compliance is related to incorrect styling of the iframe resulting from the recent routing/layout changes.  Working on options.

There three other `best practices` items that got flagged, but these are not requirements:

- Heading levels should only increase by one
    - We start with h4 in the tool/history panels, there is no h1.
- Ensure landmarks are unique
    - We have many `nav` items (content operations -- could be argued this shouldn't be a nav)
- Pages should contain a level-one heading
    - Maybe the masthead Galaxy should just be a styled h1?  I don't know that's particularly useful though.


## How to test the changes?
(Select all options that apply)
- [x] This is a refactoring of components with existing test coverage.

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

dannon committed:
Toplevel nav elements should not have aria role menuitem  (they aren't in a parent 'menu' container)

dannon committed:
Set default lang on html -- we do supply some translations and can scope those elements to other  values, but the default (and supported value) for most of the app is english.

dannon committed:
Drop a11y-breaking scale max enforcement from templates.

dannon committed:
X-UA-Compatible is no longer needed; we don't support the older browsers that this applies to anyway (and it does nothing in IE11)

dannon committed:
Fix aria role for container 'list' in history contents.
xref: https://dequeuniversity.com/rules/axe/4.4/aria-required-parent

dannon committed:
The default role here 'menu' is fine, since it's a dropdown menu.

dannon committed:
Add sr-only text to b-dropdown button in toolbox

dannon committed:
Add sr-only text to history options dropdown

dannon committed:
Add proper tooltip to history options dropdown

dannon committed:
Fix aria labels in history filter box buttons

dannon committed:
Add sr-only text to history actions cog

dannon committed:
aria-label attribute is not well supported on a div with no valid role attribute // specify menu here

dannon committed:
Fix quotameter a11y label; have to work around bootstrap-vue shortcoming
here.

xref: https://dequeuniversity.com/rules/axe/4.4/aria-progressbar-name

dannon committed:
Masthead a11y; for icon-only tabs display the tooltip as a sr-only text placeholder.

dannon committed:
Wrap app in 'main' landmark.

mvdbeek commented on client/src/components/Panels/Buttons/PanelViewButton.vue:
```diff
@@ -2,12 +2,15 @@
         aria-label="View all tool panel configurations"
         class="tool-panel-dropdown"
         size="sm">
+        <template v-slot:button-content>
```
Nice, I didn't get how that was a problem with the checker website, that should fix it.

dannon committed:
Update masthead test to check for scratchbook existence with a more specific selector.

dannon committed:
Fix historyfilters test to look for updated data description.

dannon committed:
Attribute order tweaks

dannon committed:
Add eslint plugin for vuejs accessibility testing

dannon committed:
Set refined error/warn granularity for first pass implementing vuejs-accessibility

dannon committed:
Drop usage of autofocus prop in clicktoedit
https://github.com/vue-a11y/eslint-plugin-vuejs-accessibility/blob/main/docs/no-autofocus.md

dannon committed:
Add title to galaxy viz iframe // vuejs-a11y

dannon committed:
Use tabindex 0 in sharing, flowing naturally in the page order instead
of forcing first index.

https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/tabindex#accessibility_concerns

dannon committed:
Add image alt tag to tool logo -- vue-a11y

dannon committed:
Flag two empty headers as compliant -- v-short will always insert contents here

Format.

dannon committed:
Rename CenterPanel as CenterFrame for clarity -- this is always an iframe and is too layered

mvdbeek reviewed (approved):
Still WIP ?
